src/sage/categories/modules_with_basis.py

```diff
@@ -2142,7 +2152,7 @@ def map_support_skip_none(self, f):
                 sage: y.parent() is B                                                   # needs sage.modules
                 True
             """
-            return self.parent().sum_of_terms( (fm,c) for (fm,c) in ((f(m), c) for m,c in self) if fm is not None)
+            return self.parent().sum_of_terms( (fm,c) for (fm,c) in ((f(m), c) for m,c in self.monomial_coefficients(copy=False).items()) if fm is not None)
```

Comment:
```suggestion
            mc = self.monomial_coefficients(copy=False)
            return self.parent().sum_of_terms((fm, c) for m, c in mc.items() if (fm := f(m)) is not None)
```
This can be simplified a bit.